metis/lib/server/controllers/upload_controller.rb

```diff
@@ -58,6 +52,21 @@ def upload
 
   private
 
+  def create_upload(params)
```

Comment:
So, I noticed that if a user starts a download but doesn't complete it, then tries to upload _a different file_ to the same path, the uploading fails because it re-uses the existing `Upload` model, which may have a `next_blob_hash` that doesn't match the new file that is replacing it.

Shoudl we change this behavior to more of a `create_or_reset_upload`?  If the upload does exist, update its file_by_position and next_blob* params to defaults.

This would allow starting an incomplete download over with a new file.
Should we consider changing this to a "create or reset" like behavior?  If the upload exists, it should be reset to resume from the 0 byte position.  Client'

Reply:
Hm...that's an interesting scenario. That seems like maybe a browser component problem, not a controller problem? I'll play around with it tomorrow, but wouldn't `find_or_create` get the existing `Upload` (partially completed), and then reset `next_blob_hash`?

```
    Metis::Upload.find_or_create(
      file_name: params[:file_path],
      bucket: params[:bucket],
      metis_uid: metis_uid,
      project_name: params[:project_name]
    ) do |f|
      f.author = Metis::File.author(params[:user])
      f.file_size = 0
      f.current_byte_position = 0
      f.next_blob_size = -1
      f.next_blob_hash = ''
    end
```

I wonder if the browser skips the `authorize_upload` route and instead tries uploading directly back to `upload_blob`, which I could see causing an issue with `next_blob_hash`.

Does it make sense if I make another issue around this use case, since I think it's not directly related to this specific upload-from-Timur feature, but seems like a bug somewhere?

Reply:
Since upload from Timur is to a magma-issued unique temp path resuming may be broken anyway, as the same file would goto a new path on upload retry.

As to not resuming correctly, the upload_start controller only has the file size as a hint as to what the user will send. It could reject if this mismatches but otherwise it should try to resume.

Reply:
Good point, the Timur filenames are unique for every `::temp` request. So perhaps that issue is a Metis-specific upload scenario rather than via Timur.

Reply:
Recorded in mountetna/etna#61. I think this is a client-side issue.